### ISSUE TITLE
Default schema name no longer quoted in SEARCH_PATH 

### ISSUE THREAD

**Issue by flowertwig** [liquibase/liquibase#3205]
## Environment

**Liquibase Version**:4.14.0

**Liquibase Integration & Version**: maven

**Database Vendor & Version**:postgres

## Description

The behaviour of DatabaseUtils in setting SEARCH_PATH changed in 4.14.0. It used to quote the default schema name and not the current search path. From 4.14.0 though. it quotes the current search path but not the default schema name. Since there are no tests for DatabaseUtils, I'm not sure whether this was intentional, but I noticed it since it started causing sql syntax errors for schema names containing a hyphen. I guess the intent was to quote both of them so I created a test for this.

## Steps To Reproduce

Run test: https://github.com/flowertwig/liquibase/blob/searchpath-escaping/liquibase-core/src/test/java/liquibase/database/core/DatabaseUtilsTest.java

Test results for recent versions:

4.13.0
Expected :SET SEARCH_PATH TO "defaultSchema", "currentPath"
Actual   :SET SEARCH_PATH TO "defaultSchema", currentPath

4.14.0
Expected :SET SEARCH_PATH TO "defaultSchema", "currentPath"
Actual   :SET SEARCH_PATH TO defaultSchema, "currentPath"

4.15.0
Expected :SET SEARCH_PATH TO "defaultSchema", "currentPath"
Actual   :SET SEARCH_PATH TO defaultSchema, "currentPath"

**Comment by FBurguer** [comment]
Thanks for submitting this issue! We will look into it.

**Comment by michelmeeuwissen** [comment]
Issue does not exists in 4.10.0 but started from 4.11.0+

**Comment by kferkic** [comment]
Is there any new on this?
I still have this issue with 4.17.2

**Comment by robertwheelerprodly** [comment]
I guess is there a work around or something for this?

**Comment by sjoerdschunselaar** [comment]
Yes, I found a workaround for this issue. You can set the property: liquibase.preserveSchemaCase to true.
In case you use Spring Boot, you can add:

`        System.setProperty(GlobalConfiguration.PRESERVE_SCHEMA_CASE.getKey(), "true");
`

**Comment by MaksimMyshkin** [comment]
As workaround (for Spring) I just quoted value for property `spring.liquibase.default-schema="${application.datasource.schema}"` and left unqouted for `spring.liquibase.liquibase-schema=${application.datasource.schema}`. As raw liquibase properties I guess it be default-schema-name and liquibase-schema-name respectively.

**Comment by anton-pozhidaev** [comment]
Just additional 50 cents from my case probably it will be useful. 
If name of app the same of schema it works good without _spring.liquibase.default-schema_ property. I have few services working fine with that config (with _spring.liquibase.liquibase-schema_ only) but they were started like this initially.

But why I'm here? I have service with name "orders" connected to schema "order" and after liquibase 4.13.0 app does't start because topic issue. After schema has renamed to "orders" and added spring.liquibase.default-schema property - it's working good with last version. But don't start without _spring.liquibase.default-schema_

**Comment by psavva** [comment]
I'm also running into this issue with Liquibase Version 4.22.0
I'm using spring boot, and have not managed to get any of the workarounds to work correctly.
Firstly, Spring Boot does not support all liquibase parameters out the box, so I implemented a Component for it instead, no dice...

import jakarta.annotation.PostConstruct;
import liquibase.GlobalConfiguration;
import org.springframework.stereotype.Component;

```
@Component
public class LiquibaseConfig {
    @PostConstruct
    public void setProperty() {
        System.setProperty(GlobalConfiguration.PRESERVE_SCHEMA_CASE.getKey(), "true");
    }
}
```

**Comment by fatso83** [comment]
@psavva You need to set that property before the Spring stuff comes into play. Worked just fine in our Spring Boot 2.7 and SB 3 apps.

**Comment by DmitryLobachov** [comment]
I also encountered this issue during the migration to Spring Boot 3.
Any updates on it?

**Comment by fatso83** [comment]
@dimon1165 Obviously not, but there is a workaround which will give you the old behavior back. As that workaround exists, I would not assume they would prioritize this very highly. I would expect this is the new behavior.

**Comment by fatso83** [comment]
@FBurguer This is so annoying that I thought I might have a stab at this. Is there some obvious really-hard-to-fix reason why you guys haven't touched this that I should be aware of, or is this easy pickings?

**Comment by yelhouti** [comment]
FYI: 
```
 System.setProperty(GlobalConfiguration.PRESERVE_SCHEMA_CASE.getKey(), "true");
```
didn't totally fix it for me, I was using 
```
<sql>UPDATE ${database.defaultSchemaName}.customer...</sql>
```
In some migration scripts and since I didn't quote them here they failed. 
Also try to use the previous checksum in the changeSet with:
```
<validCheckSum>8:66de2e69d671eac9cbed08f7a33debf5</validCheckSum>
```
didn't work great two as postgres and h2 have different checksums for sql statements, so I decide to just start blocking hyphens...

**Comment by magicus** [comment]
I don't know exactly when this was solved, but it does not seem to be a problem anymore in 4.33.0 at least.

**Comment by mirvnillith** [comment]
We're running 4.33.0 and seeing unquoted (and thus failing) calls: SET SEARCH_PATH TO peppol-id-manager, "$user","public" (dash causes a syntax error)

EDIT: Seems we only get this on Windows and not Unix. The PRESERVE_SCHEMA_CASE-workaround works (for us) on Windows and that's good enough. But it's probably not a good idea to have a platform-dependent issue like this?

**Comment by thecoolestcoderrushil** [comment]
An unquoted schema name in SET SEARCH_PATH is a painful failure, since schemas with hyphens break and the regression has persisted across versions. Rehearsing the migration on a production-shaped twin first would surface the quoting problem before deploy day. antifailure.dev (https://antifailure.dev) spins up disposable database twins for migration rehearsal, and it is open source and free to try. Happy to set up a demo if that would help.
